### PR TITLE
Login page using Keycloak 

### PR DESCRIPTION
**Pull request by binni979**
This is a pull request to solve [this](https://github.com/NASA-IMPACT/stac-admin/issues/2) issue. 

When the user goes to the URL, they're taken to a login page (provided by keycloak). 
Screenshot 2024-08-14 at 2 48 36 PM

Once the user logs in successfully, they can access the stac admin features.
Screenshot 2024-08-14 at 2 49 04 PM

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -20,9 +23,37 @@ import ItemForm from "./pages/ItemForm";
 import NotFound from "./pages/NotFound";
 import CollectionDetail from "./pages/CollectionDetail";
 
-export const App = () => (
+// Keycloak configuration
+const keycloakInstance = new Keycloak({
+  realm: "stac-realm",
+  url: "http://localhost:8080/",  // Ensure the URL is correct
+  clientId: "stac-admin",
```
Comment (slesaad) [inline src/App.tsx]: these should come from environment variables (.env)

File: src/components/ItemResults/MapView.tsx [hunk 2]
```diff
@@ -1,6 +1,7 @@
 import { useCallback, useEffect, useMemo, useState } from "react";
 import { Box } from "@chakra-ui/react";
-import Map, {  type MapRef, type MapLayerMouseEvent, Source, Layer } from "react-map-gl/maplibre";
+// import Map, {  type MapRef, type MapLayerMouseEvent, Source, Layer } from "react-map-gl/maplibre";
+import Map, { MapRef, MapLayerMouseEvent, Source, Layer } from "react-map-gl/maplibre";
```
Comment (slesaad) [inline src/components/ItemResults/MapView.tsx]: why this change?

File: src/index.tsx [hunk 3]
```diff
@@ -1,29 +1,26 @@
...
+// );
 root.render(
-  <React.StrictMode>
+  <>
     <ColorModeScript />
-    <App />
-  </React.StrictMode>,
+    <WrappedApp />
+  </>,
 );
```
Comment (slesaad) [inline src/index.tsx:23]: why remove `React.StrcitMode` ?

**Comment by slesaad** [comment]
From https://github.com/NASA-IMPACT/stac-admin/issues/2:
> All API requests will send the token obtained from keycloak as an authentication parameter in the header.

I don't see this in the code?

File: src/index.tsx (hunk 3 above)
Comment (binni979) [reply src/index.tsx:23]: When I used `React.StrcitMode` I got this error 
Screenshot 2024-08-15 at 4 35 05 PM
so, to solve it I remove that.

File: src/App.tsx (hunk 1 above)
Comment (binni979) [reply src/App.tsx]: added environment variables

File: src/components/ItemResults/MapView.tsx (hunk 2 above)
Comment (binni979) [reply src/components/ItemResults/MapView.tsx]: converted into 
`import Map, {  type MapRef, type MapLayerMouseEvent, Source, Layer } from "react-map-gl/maplibre";`

File: src/keycloak.ts [hunk 4]
```diff
@@ -3,6 +3,6 @@ import Keycloak from "keycloak-js";
 
 export const keycloakInstance = new Keycloak({
   realm: process.env.REACT_APP_KEYCLOAK_REALM,
-  url: process.env.REACT_APP_KEYCLOAK_WORKFLOWS_API,
+  url: process.env.REACT_APP_KEYCLOAK_URL,
```
Comment (slesaad) [inline src/keycloak.ts]: why?

Comment (binni979) [reply src/keycloak.ts]: I made changes to the previous one `REACT_APP_KEYCLOAK_WORKFLOWS_API`.

Comment (slesaad) [reply src/keycloak.ts]: why are you changing this? this doesn't need to be changed?

Comment (binni979) [reply src/keycloak.ts]: I mistakenly changed that, sorry.
